src/components/ProgressBar/index.js

```diff
@@ -0,0 +1,128 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+const ProgressBar = (props) => {
+  const {
+    value,
+    max,
+    theme,
+    size,
+    label,
+    showPercentage,
+    percentagePosition,
+    percentageText,
+    className,
+    id,
+    ...rest
+  } = props;
+
+  const percentage = Math.min(Math.max(0, (value / max) * 100), 100);
+
+  const getCssClasses = () => {
+    const cssClasses = ["tyk-progress-bar"].concat(
+      className ? className.split(" ") : []
+    );
+    const sizeClass = size
+      ? `tyk-progress-bar--${size}`
+      : "tyk-progress-bar--md";
+    const themeClass = theme
+      ? `tyk-progress-bar--${theme}`
+      : "tyk-progress-bar--primary";
+
+    cssClasses.push(sizeClass);
+    cssClasses.push(themeClass);
+
+    return cssClasses.join(" ");
+  };
+
+  const renderPercentage = () => {
+    if (!showPercentage) return null;
+
+    const percentageValue = Math.round(percentage);
+    const text = percentageText
+      ? percentageText.replace("{value}", percentageValue)
+      : `${percentageValue}%`;
```

Comment:
Replacing pieces of text won't work well with i18n. If this is needed I think a function that is given the value and returns a text should be used instead.